### PR TITLE
Extract out ShowCodeToggle component from CodeWorkspace

### PR DESCRIPTION
**Pull request by pcardune**
Another fix, this time to handle cases where ShowCodeToggle is somehow rendered _before_ StudioApp.init gets called.

### CODE REVIEW

**Comment by islemaster** [comment]
This should be the root cause of the toggle showing up where it shouldn't (for example, Jigsaw levels)?

**Comment by pcardune** [comment]
as far as i can tell. I checked each of the pages that you screenshotted in the previous PR and they all worked as intended.

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ import project from './code-studio/initApp/project';
 import * as assets from './code-studio/assets';
 import i18n from './code-studio/i18n';
 import AbuseError from './code-studio/components/abuse_error';
+import ObservableEvent from './ObservableEvent';
```
Comment (islemaster) [inline apps/src/StudioApp.js]: Not sure which I prefer, but do you have thoughts on using our own `ObservableEvent` vs Node.js' [`EventEmitter`](https://nodejs.org/api/events.html) (which we're [using in Maker](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/kits/maker/CircuitPlaygroundBoard.js#L36))?

Comment (pcardune) [reply apps/src/StudioApp.js]: I definitely prefer EventEmitter. Didn't realize we were already using that. I looked in `package.json` but didn't see the event emitter library.

Comment (islemaster) [reply apps/src/StudioApp.js]: Yeah, we get it through node-libs-browser?
```jsx
import {EventEmitter} from 'events'; // provided by webpack's node-libs-browser
```

**Pull request merged**